{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"LiaoSteve.ipynb","provenance":[],"authorship_tag":"ABX9TyMiSWPQQp9Q1DJ+89kqDhA4"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"code","metadata":{"id":"0ss_wGTAO9Zz"},"source":["!git clone https://github.com/LiaoSteve/Trash-dataset-and-object-detection.git"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"MD2U8LoMPJHW"},"source":["cd Trash-dataset-and-object-detection/ "],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"sQLG1e8aR6EQ"},"source":["!python train.py"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"cZx7uRkxSD5L"},"source":["!wget https://pjreddie.com/media/files/yolov3.weights"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"yfEnNOPzSuZJ"},"source":["!python convert.py yolov3.cfg yolov3.weights model_data/yolo.h5"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"3dpQcjx8TFb1"},"source":["!python yolo_video.py"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"QiyoAAm3TOJk"},"source":["!pip install git+git://github.com/fchollet/keras.git --upgrade"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"FQG9l2zKTddO"},"source":["!pip install -U setuptools"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"MtTGXwLHToPS"},"source":["!pip install tensorflow-estimator"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"W5dwKCcJT1ES"},"source":["pip install keras"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"6BRsgnmoT5Bn","executionInfo":{"status":"ok","timestamp":1611406549522,"user_tz":-390,"elapsed":7077,"user":{"displayName":"Yuzana Win","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhSaiXVwKq1GHHcf9s_Y69gz9TbF75_v-tpGpA5FRE=s64","userId":"14284865208094345368"}},"outputId":"fc7ab8ba-94bf-4308-a1b4-95df65ef2a7a"},"source":["!python train.py"],"execution_count":12,"outputs":[{"output_type":"stream","text":["2021-01-23 12:55:42.788117: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1\n","2021-01-23 12:55:45.078128: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","2021-01-23 12:55:45.079452: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1\n","2021-01-23 12:55:45.090377: E tensorflow/stream_executor/cuda/cuda_driver.cc:328] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected\n","2021-01-23 12:55:45.090445: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (a8edce38c3e7): /proc/driver/nvidia/version does not exist\n","2021-01-23 12:55:45.091036: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set\n","Create YOLOv3 model with 9 anchors and 3 classes.\n","Traceback (most recent call last):\n","  File \"train.py\", line 190, in <module>\n","    _main()\n","  File \"train.py\", line 33, in _main\n","    freeze_body=2, weights_path='model_data/yolo_weights.h5') # make sure you know what you freeze\n","  File \"train.py\", line 120, in create_model\n","    model_body.load_weights(weights_path, by_name=True, skip_mismatch=True)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py\", line 2227, in load_weights\n","    with h5py.File(filepath, 'r') as f:\n","  File \"/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py\", line 408, in __init__\n","    swmr=swmr)\n","  File \"/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py\", line 173, in make_fid\n","    fid = h5f.open(name, flags, fapl=fapl)\n","  File \"h5py/_objects.pyx\", line 54, in h5py._objects.with_phil.wrapper\n","  File \"h5py/_objects.pyx\", line 55, in h5py._objects.with_phil.wrapper\n","  File \"h5py/h5f.pyx\", line 88, in h5py.h5f.open\n","OSError: Unable to open file (unable to open file: name = 'model_data/yolo_weights.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"WPBc72aoUIY4"},"source":[""],"execution_count":null,"outputs":[]}]}